src/main/java/com/amazon/opendistroforelasticsearch/sql/antlr/semantic/visitor/TypeChecker.java

```diff
@@ -173,6 +173,11 @@ public Type visitNull() {
         return UNKNOWN;
     }
 
+    @Override
+    public Type visitConvertedType(String text) {
+        return ESDataType.typeOf(text);
```

Comment:
David's PR reminds me that some type in `convertedDataType` doesn't have corresponding type in `ESDateType`, such as `DATETIME` and `INT`. Could you verify if it's an issue or not? I guess `UNKNOWN` is returned in this case. It is fine for type checking but it may be not what you expect.

Reply:
Sure, I will add the default type for UNKNOWN use cases when applying the return type.